edx_django_utils/hooks/pipeline.py

```diff
@@ -0,0 +1,90 @@
+"""
+TODO: add description
+"""
+from logging import getLogger
+
+from .exceptions import HookException
+from .utils import get_functions_for_pipeline, get_hook_configurations
+
+log = getLogger(__name__)
+
+
+def run_filter_pipeline(trigger_name, *args, **kwargs):
+    """
+    Function that manages the filters pipeline.
+
+    Arguments:
+        trigger_name:
+
+    Returns:
+        result:
+    """
+    result = None
+
+    hook_config = get_hook_configurations(trigger_name)
+
+    pipeline = hook_config.get("pipeline", [])
```

Comment:
should we consider empty pipelines? @morenol

Reply:
By default they should be empty, so I guess that yes. It is possible

Reply:
But should we call _run_pipeline with an empty pipeline?

Reply:
No. We could return the kwargs in that case in order to keep consistency with the current implementation of the pipeline.

Reply:
Thinking about this, do we need to return something from the pipeline? Maybe the last return is not needed. Do we have an use case where that could be needed?

Reply:
The `Openedx hooks location` comments section for the Hook `openedx.lms.auth.pre_login.filter.v1` says that we need to run all hooks and check which one denies the login action. How can we check this if the pipeline does not have a return value?

Reply:
With the exception, if any action raises the HookException then that would deny the login.

Reply:
What about row 44 from the same document? Where we extend the certificate template

Reply:
I think that for contexts we should use the tools that are already in the plugin infrastructure